core/kernel/tee_ta_manager.c

```diff
@@ -626,7 +627,7 @@ static void update_current_ctx(struct thread_specific_data *tsd)
 	if (((ctx && is_user_ta_ctx(ctx) ?
 			to_user_ta_ctx(ctx)->mmu : NULL) == NULL) ==
 					core_mmu_user_mapping_is_active())
-		panic();
+		panic("unexpected active mapping");
```

Comment:
actually these where found at build time. just that i did push without
testing....
oups!

i'll push updated change soon.
I tried playing around with panic()/panic("string") but still no solution.

On 15 July 2016 at 14:51, Jérôme Forissier <notifications@github.com> wrote:

> In core/kernel/tee_ta_manager.c
> <https://github.com/OP-TEE/optee_os/pull/910#discussion_r70967244>:
>
> > @@ -626,7 +627,7 @@ static void update_current_ctx(struct thread_specific_data *tsd)
> >  	if (((ctx && is_user_ta_ctx(ctx) ?
> >  			to_user_ta_ctx(ctx)->mmu : NULL) == NULL) ==
> >  					core_mmu_user_mapping_is_active())
> > -		panic();
> > +		panic("unexpected active mapping");
>
> You should replace #define panic(...) __do_panic(__FILE__, __LINE__,
> __func__, (void *)0) with #define panic() __do_panic(__FILE__, __LINE__,
> __func__, (void *)0) so that those mistakes are detected at compile time.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OP-TEE/optee_os/pull/910/files/425c32dfd8a53945f7016ce3dcdf713c5bd899f5..2314c9d027ebc3e128c2495501c50366851858a9#r70967244>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATEHJmqdZel5ruYDo7Y6lqeyHZCFf-leks5qV4JTgaJpZM4JJ0Oa>
> .
>